pkg/webhookconfig/configmanager_test.go

```diff
@@ -0,0 +1,95 @@
+package webhookconfig
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+var (
+	emptyInternalRules []interface{}
+	emptyAPIRules      []interface{}
+
+	configmapsInternalRules []interface{}
+	configmapsAPIRules      []interface{}
+
+	configmapsSecretsInternalRules []interface{}
+	configmapsSecretsAPIRules      []interface{}
+)
+
+func init() {
+	// No rules.
+	// Internal representation is a rule with no selectors.
+	// API server representation is no rule (nil).
+	emptyInternalRules = []interface{}{map[string]interface{}{}}
+	emptyAPIRules = nil
+
+	// Rule selecting configmaps.
+	// API server representation matches the internal
+	// representation but has extra fields "operations" and "scope".
+	configmapsInternalRules = []interface{}{
+		map[string]interface{}{
+			"apiGroups":   []string{""},
+			"apiVersions": []string{"v1"},
+			"resources":   []string{"configmaps"},
+		},
+	}
+	configmapsAPIRules = []interface{}{map[string]interface{}{
+		"apiGroups":   []interface{}{""},
+		"apiVersions": []interface{}{"v1"},
+		"resources":   []interface{}{"configmaps"},
+		"operations":  []interface{}{"CREATE", "UPDATE", "DELETE", "CONNECT"},
+		"scope":       "*",
+	}}
+
+	// Rule selecting configmaps and secrets.
+	// API server representation matches the internal
+	// representation but has extra fields "operations" and "scope".
+	configmapsSecretsInternalRules = []interface{}{
+		map[string]interface{}{
+			"apiGroups":   []string{""},
+			"apiVersions": []string{"v1"},
+			"resources":   []string{"configmaps", "secrets"},
+		},
+	}
```

Comment:
Can we add another test case to test the ordering of the resources?
```
map[string]interface{}{
			"apiGroups":   []string{""},
			"apiVersions": []string{"v1"},
			"resources":   []string{"secrets", "configmaps"},
		}
```

Reply:
I can add the test, but are you expecting it to return equal or not equal to the other way around?
The end result for the webhook is the same, but trying to account for that here makes things more complicated.
And as I understand it
* The API server does not rearrange the order of the list
* The resources in the webhook generated by kyverno should be deterministic (but I'm not certain).

But it would be a problem if the generated ordering was changing.

Reply:
>but are you expecting it to return equal or not equal to the other way around?

I was trying to understand the behavior, it should be "not equal" to me.

>The resources in the webhook generated by kyverno should be deterministic (but I'm not certain)

It could be different based on returned policies when listing. For example, if you have three policies that match pods, jobs, and pods separately, the order should be []{"pods", "jobs"}; but if the first policy is removed, then the order becomes []{"jobs", "pods"}.

Reply:
I've added a test for that, ensuring not equal.
If it still causes the webhook to be always updated, but only switching the order of the selectors, then I can look at fixing that (by sorting the values maybe).